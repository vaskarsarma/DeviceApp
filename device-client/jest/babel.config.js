const babelJest = require('babel-jest');

const config = {
  presets: ['react', 'stage-2', 'env'],
  retainLines: true,
  plugins: ['transform-decorators-legacy', 'transform-class-properties', 'add-module-exports', 'lodash'],
};

module.exports = babelJest.createTransformer(config);
