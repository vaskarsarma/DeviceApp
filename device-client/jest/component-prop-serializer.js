const { Component } = require('react');

// Fix snapshots break in case of component reference passed as a prop
module.exports = {
  test(val) {
    return val instanceof Component;
  },
  print() {
    return '[Component]';
  },
};
