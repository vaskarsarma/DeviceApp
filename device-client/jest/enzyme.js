/* eslint-disable no-underscore-dangle */
import Enzyme, {
  configure, shallow, mount, render,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
export { shallow, mount, render };
export default Enzyme;

global.window._env_ = {};
