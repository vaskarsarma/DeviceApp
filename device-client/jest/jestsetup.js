/* eslint-disable global-require */
import 'babel-regenerator-runtime';
import noop from 'lodash/noop';
import React from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.3';

process.env.TEST = true;
process.env.UI_ASSETS_URL = '/uiassets/';
global.__SAFE_RENDER__ = false;
// apply React 16 adapter
Enzyme.configure({ adapter: new Adapter() });

// Enzyme methods save to global
global.shallow = shallow;
global.render = render;
global.mount = mount;
global.MEDIA_URL = '';
global.__BROWSER__ = true;
global.__APP_MOUNT__ = true;
global.__INITIAL_STATE__ = { get resources() {
  return global.resources;
}};

jest.mock('store', () => {
  global.resources = setResources();

  function setResources(newresources = {}) {
    global.resources = {
      ...newresources,
      getRawResource(string) {
        return global.resources[string];
      },
    };
  }

  function setResMap(map) {
    if (!map) {
      return {};
    }

    const { resources } = global;
    return Object.keys(map).reduce((res, key) => {
      if (typeof map[key] !== 'object') {
        res[key] = resources[map[key]];
      } else if (Array.isArray(map[key])) {
        res[key] = map[key].map(item => setResMap(item));
      } else {
        res[key] = setResMap(map[key]);
      }
      return res;
    }, {});
  }

  return {
    ...require.requireActual('store'),
    __setResources: jest.fn(setResources),
    connectResources: jest.fn(map => Comp => {
      if (!global.resources) {
        setResources();
      }
      function Wrapper(props) {
        const resMap = {
          ...setResMap(map),
          getRawResource(string) {
            return global.resources[string];
          },
        };
        return (
          <Comp
            {...props}
            resources={resMap}
          />
        );
      }
      Wrapper.displayName = `WithResources(${Comp.displayName || Comp.name})`;
      return Wrapper;
    }),

  };
});

// eslint-disable-next-line global-require
jest.mock('react-responsive-redux', () => {
  let fakeWidth = 1240;
  return {
    __setFakeWidth: value => (fakeWidth = value),
    responsiveWrapper: jest.fn(({ minWidth = 0, maxWidth = Infinity }) => ({ children }) =>
      (fakeWidth >= minWidth && fakeWidth <= maxWidth ? children : null),
    ),
  };
});

jest.mock('react-dom', () => {
  const __portalNodes = [];
  return {
    ...require.requireActual('react-dom'),
    __portalNodes,
    createPortal: jest.fn((el, node) => {
      if (!__portalNodes.find(portalNode => node === portalNode)) {
        __portalNodes.push(node);
      }
      return el;
    }),
  };
});

jest.mock('utils/dom-utils/get-modal-root', () => {
  const modalRoot = {
    appendChild: jest.fn(),
    removeChild: jest.fn(),
    querySelector: jest.fn(),
    querySelectorAll: jest.fn(),
    clientWidth: 0,
  };
  return {
    getModalRoot: () => modalRoot,
  };
});

jest.mock('utils/dom-utils/ping-image-url', () => jest.fn(src => Promise.resolve(src)));

global.Element.prototype.querySelector = jest.fn(() => ({
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
  focus: jest.fn(),
  setAttribute: jest.fn(),
  getBoundingClientRect: jest.fn(() => ({
    height: 0,
    width: 0,
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
  })),
  style: jest.fn(() => {}),
}));

const _Date = Date;
global.MOCK_TIME = 1514764800000;
global.Date = jest.fn((...args) => (args.length ? new _Date(...args) : new _Date(global.MOCK_TIME)));
global.Date.now = jest.fn(() => global.MOCK_TIME);
global.Date.UTC = _Date.UTC;

jest.mock('@atoms/script', () => jest.fn(() => null));
jest.mock('utils/hocs/with-breakpoints', () => {
  const actual = require.requireActual('utils/hocs/with-breakpoints');
  const __contextDataSetter = jest.fn(() => ({ breakpoint: 'medium' }));
  const mock = jest.fn(() => Component => props => (
    <Component
      {...__contextDataSetter()}
      {...props}
    />
  ));
  mock.BREAKPOINTS_NAMES = actual.default.BREAKPOINTS_NAMES;
  mock.BreakpointPropType = actual.BreakpointPropType;
  mock.__contextData = __contextDataSetter;
  return mock;
});

const createStorage = () => {
  let store = {};
  return {
    getItem(key) {
      return store[key] || null;
    },
    setItem(key, data) {
      store[key] = data.toString();
    },
    removeItem(key) {
      delete store[key];
    },
    clear() {
      store = {};
    },
  };
};

global.dataLayer = { push: noop };
global.localStorage = createStorage();
global.sessionStorage = createStorage();

const { createEvent } = global.Document.prototype;
global.document.createEvent = function _createEvent(type, ...arg) {
  const typeLower = type.toLowerCase();

  if (typeLower === 'storageevent') {
    const storageevent = createEvent.call(global.document, 'customevent', ...arg);
    storageevent.initStorageEvent = (name, bubbles, cancelable) => {
      storageevent.initEvent(name, bubbles, cancelable);
    };
    return storageevent;
  }

  return createEvent.call(global.document, type, ...arg);
};

jest.mock('store/selectors', () => {
  const __resourcesGetter = jest.fn(() => ({}));
  const actual = require.requireActual('store/selectors');
  const resourcesSelector = jest.fn((...args) => __resourcesGetter(...args));
  return {
    ...actual,
    __resourcesGetter,
    __resourcesDefaultSelector: actual.resourcesSelector,
    __useActualResourceSelector: () => resourcesSelector.mockImplementation(actual.resourcesSelector),
    __useMockResourceSelector: () => resourcesSelector.mockImplementation(__resourcesGetter),
    resourcesSelector,
  };
});
