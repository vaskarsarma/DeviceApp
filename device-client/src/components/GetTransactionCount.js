import React, { Component } from 'react';
import { getTransactionCount } from "../api";
import { Form, Button } from 'react-bootstrap';
import openSocket from 'socket.io-client';
import { formatDateandAddOneDay } from '../utils/utils';

export default class GetTransactionCount extends Component {

  constructor() {
    super();
    this.state = {
      transactionInfo: [],
      count: 0,
      fromDate: '2021-08-27',
      toDate: formatDateandAddOneDay(new Date()),
    }
  }

  componentDidMount() {
    this.getTransactions();

    // Open Socket connection with Server
    const apidomain = window._env_.REACT_APP_APIDOMAIN;
    const socket = openSocket(apidomain);
    socket.on('addtempstats', data => {
      if (data.action === 'add') {
        console.log("Update temperature stats triggered");
        this.getTransactions();
      }
    });

    socket.on('deletetempstats', data => {
      if (data.action === 'delete') {
        console.log("Update temperature stats triggered");
        this.getTransactions();
      }
    });
  }

  handleOnSubmit = (e) => {
    e.preventDefault();
    console.log("Clicked the submit");
    this.getTransactions();
  }

  handleInputChange = (e) => {
    const { name, value } = e.target;
    this.setState({
      [name]: value
    });
    console.log(`Updated : ${e.target.name}, ${e.target.value}`);
  }

  getTransactions() {
    getTransactionCount(this.state.fromDate, this.state.toDate).then((results) => {
      if (results !== null) {
        this.setState({
          transactionInfo: results,
          count: results.length
        });
      }
    });
  }

  render() {
    let transactionInfo = (<p>View max/min transactions count filtered by DataRange</p>);

    if (this.state.count > 0) {
      transactionInfo = (<div>
        <p>View max/min transactions count filtered by DataRange</p>
        <table className="table table-bordered">
          <thead>
            <tr>
              <th scope="col">#</th>
              <th scope="col">Device Name</th>
              <th scope="col">Transaction Count</th>
            </tr>
          </thead>
          <tbody>
            {this.state.transactionInfo.map((p, i) => {
              return (
                <tr key={p._id} dataid={p._id}>
                  <th scope="row">{(i === 0) ? 'Device with max transactions' : 'Device with min transactions'}</th>
                  <td>{p._id}</td>
                  <td>{p.count}</td>
                </tr>
              )
            })
            }
          </tbody>
        </table>
      </div>
      )
    }

    return (
      <React.Fragment>
        <Form onSubmit={this.handleOnSubmit} className="form-inline">
          <Form.Group controlId="fromDate" className="form-group mb-2">
            <Form.Label className="sr-only">From</Form.Label>
            <Form.Control
              className="form-control"
              type="Date"
              name="fromDate"
              value={this.state.fromDate}
              placeholder="Enter from date"
              required={true}
              onChange={this.handleInputChange}
            />
          </Form.Group>
          <Form.Group controlId="toDate" className="form-group mb-2">
            <Form.Label className="sr-only">To</Form.Label>
            <Form.Control
              className="form-control"
              type="Date"
              name="toDate"
              value={this.state.toDate}
              placeholder="Enter to date"
              required={true}
              onChange={this.handleInputChange}
            />
          </Form.Group>
          <Button variant="primary" type="submit" className="btn btn-primary mb-2">
            Submit
          </Button>
        </Form>
        <div className="device-stats-list">
          {transactionInfo}
        </div>
      </React.Fragment>
    );
  }
}

