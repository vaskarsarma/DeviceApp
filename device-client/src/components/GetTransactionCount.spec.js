/* eslint-disable import/no-named-as-default-member */
/* eslint-disable import/no-named-as-default */
import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from '../../jest/enzyme';
import GetTransactionCount from './GetTransactionCount';

describe('<GetTransactionCount />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<GetTransactionCount/>);
  });

  it('Render Correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
