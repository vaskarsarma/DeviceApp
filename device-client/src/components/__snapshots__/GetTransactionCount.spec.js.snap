// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GetTransactionCount /> Render Correctly 1`] = `
<Fragment>
  <Form
    className="form-inline"
    inline={false}
    onSubmit={[Function]}
  >
    <FormGroup
      className="form-group mb-2"
      controlId="fromDate"
    >
      <FormLabel
        className="sr-only"
        column={false}
        srOnly={false}
      >
        From
      </FormLabel>
      <FormControl
        className="form-control"
        name="fromDate"
        onChange={[Function]}
        placeholder="Enter from date"
        required={true}
        type="Date"
        value="2021-08-27"
      />
    </FormGroup>
    <FormGroup
      className="form-group mb-2"
      controlId="toDate"
    >
      <FormLabel
        className="sr-only"
        column={false}
        srOnly={false}
      >
        To
      </FormLabel>
      <FormControl
        className="form-control"
        name="toDate"
        onChange={[Function]}
        placeholder="Enter to date"
        required={true}
        type="Date"
        value="2021-09-02"
      />
    </FormGroup>
    <Button
      active={false}
      className="btn btn-primary mb-2"
      disabled={false}
      type="submit"
      variant="primary"
    >
      Submit
    </Button>
  </Form>
  <div
    className="device-stats-list"
  >
    <p>
      View max/min transactions count filtered by DataRange
    </p>
  </div>
</Fragment>
`;
